app/src/main/java/chat/rocket/android/util/extensions/View.kt

```diff
@@ -7,15 +7,27 @@ import android.view.View
 import chat.rocket.android.R
 import timber.log.Timber
 
-fun View.openTabbedUrl(url: Uri) {
+fun View.openTabbedUrl(url: String) {
     with(this) {
+        val uri = url.ensureScheme()
         val tabsbuilder = CustomTabsIntent.Builder()
         tabsbuilder.setToolbarColor(ResourcesCompat.getColor(context.resources, R.color.colorPrimary, context.theme))
         val customTabsIntent = tabsbuilder.build()
         try {
-            customTabsIntent.launchUrl(context, url)
+            customTabsIntent.launchUrl(context, uri)
         } catch (ex: Exception) {
             Timber.d(ex, "Unable to launch URL")
         }
     }
+}
+
+private fun String.ensureScheme(): Uri? {
+    // check if the URL starts with a http(s) scheme
+    val url = if (!this.matches(Regex("^([h|H][t|T][t|T][p|P]).*"))) {
+        "http://$this"
```

Comment:
What if the scheme is HTTPS?

Reply:
This is just to check if the url already has a scheme... Sometimes we receive urls like "www.something.com", so we append "http://"

Reply:
Gotcha! 👍